### PR TITLE
Don't write unknown case hashtags to xml

### PR DESCRIPTION
**Pull request by emord**
Only write "valid" hashtags to the xml (where valid is a "known" hashtag conversion)

@orangejenny @snopoke

### CODE REVIEW

**Comment by emord** [comment]
Updated after offlining with @orangejenny 

It also removes any case properties from an internal vellum xml tag that are no longer sent down by HQ

File: src/form.js [hunk 1]
```diff
@@ -185,6 +185,12 @@ define([
         removeHashtag: function(hashtag) {
             delete this.hashtagDictionary[hashtag];
         },
+        clearNullHashtags: function () {
+            this.hashtagDictionary = _.chain(this.hashtagDictionary)
```
Comment (orangejenny) [inline src/form.js]: Someday underscore will release a `filterObject` method so this can be one function call instead of four. But not today.

**Comment by snopoke** [comment]
Build failing
```

  1) The data browser when loaded before the form should not write unknown referenced hashtags to form:
      AssertionError: expected { '#case/dob': null } to equal { '#case/dob': null }
      + expected - actual
      
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:210
      at assertEqual (file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:782)
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:4192
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:2329
      at file:///home/travis/build/dimagi/Vellum/tests/databrowser.js:188
  2) The data browser when loaded after the form with loaded xml should not write unknown referenced hashtags to form:
      AssertionError: expected { '#case/dob': null } to equal { '#case/dob': null }
      + expected - actual
      
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:210
      at assertEqual (file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:782)
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:4192
      at file:///home/travis/build/dimagi/Vellum/bower_components/chai/chai.js:2329
      at file:///home/travis/build/dimagi/Vellum/tests/databrowser.js:397
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3305
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3148
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3260
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3350
      at file:///home/travis/build/dimagi/Vellum/_build/src/local-deps.js:16
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3305
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3148
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3260
      at file:///home/travis/build/dimagi/Vellum/bower_components/jquery/dist/jquery.js:3350
      at file:///home/travis/build/dimagi/Vellum/_build/src/local-deps.js:16
```

File: src/form.js (hunk 1 above)
Comment (emord) [inline src/form.js]: https://lodash.com/docs#omitBy

Was not trivial to migrate teh last time I looked at it, but that was when we were on 1.4

https://github.com/lodash/lodash/wiki/Migrating that's probably from 1.8 to lodash. not sure how feasible it is to migrate to 1.8, but would be happy if we did :)

Comment (orangejenny) [inline src/form.js]: Hmm...I would not support switching to lodash unless either we also switched HQ or there's something desperately awesome about lodash. Would support upgrading underscore to 1.8, though.

**Comment by emord** [comment]
:white_check_mark:

**Pull request merged**